### PR TITLE
refactor(client): social plugin

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- extract plugin protocol for further integration of 3rd party native SDKs
- implement `LogtoSocialPluginWeb` for OAuth social identity providers

## Testing

tested in demo - GitHub social works

## Reviewers

@logto-io/eng

### CODE REVIEW

File: Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthSession.swift [hunk 1]
```diff
@@ -31,12 +32,14 @@ public class LogtoWebViewAuthSession: NSObject {
...
     let redirectUri: URL
+    let socialPlugins: [LogtoSocialPlugin]
     let onFinish: FinishHandler
     var viewController: LogtoWebViewAuthViewController?
 
-    public init(_ uri: URL, redirectUri: URL, onFinish: @escaping FinishHandler) {
+    public init(_ uri: URL, redirectUri: URL, socialPlugins: [LogtoSocialPlugin], onFinish: @escaping FinishHandler) {
         self.uri = uri
         self.redirectUri = redirectUri
+        self.socialPlugins = socialPlugins
```
Comment (demonzoo) [inline Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthSession.swift:42]: Should handle these uncovered code lines?

image

Comment (gao-sun) [reply Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthSession.swift:42]: will add tests later LOG-1525

File: Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift [hunk 2]
```diff
@@ -70,36 +71,23 @@ extension LogtoWebViewAuthViewController: WKScriptMessageHandler {
...
+        }) else {
+            // TO-DO: error handling
+            return
         }
 
-        session.prefersEphemeralWebBrowserSession = true
-        session.presentationContextProvider = self
-        session.start()
+        socialPlugin
+            .start(LogtoSocialPluginConfiguration(redirectUri: redirectUri, callbackUri: callbackUri,
```
Comment (simeng-li) [inline Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift:88]: nit: not sure about the swift practice, will it be better to define a LogtoSocialPluginConfiguration variable here.

**Review by simeng-li — APPROVED** [review]
LGTM

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Review by wangsijie — APPROVED** [review]

File: Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift (hunk 2 above)
Comment (gao-sun) [reply Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift:88]: i think it's okay for now - let's define the var when it has more params

**Pull request merged**